packages/snage/src/config/parser.test.ts

```diff
@@ -26,4 +26,56 @@ describe('parseConfig', () => {
             })
         );
     });
+    it('fieldname containing non existing fields is invalid', () => {
+        const config = {
+            filename: '${banana}.md',
+            fields: [{name: 'version', type: 'semver'}],
+            standard: {
+                query: '',
+                sort: {
+                    field: 'version',
+                    order: 'desc',
+                },
+            },
+        };
+        expect(parseConfig(config)).toMatchObject(left({msg: 'Error in filename - Found reference to not existing field in file name: banana'}));
+    });
+    it('fieldname containing optional field is invalid', () => {
+        const config = {
+            filename: '${likesBanana}.md',
+            fields: [
+                {name: 'version', type: 'semver'},
+                {name: 'likesBanana', type: 'boolean', optional: true},
+            ],
+            standard: {
+                query: '',
+                sort: {
+                    field: 'version',
+                    order: 'desc',
+                },
+            },
+        };
+        expect(parseConfig(config)).toMatchObject(
+            left({msg: 'Error in filename - Fields used in the file name must not be optional or lists: likesBanana'})
+        );
+    });
+    it('fieldname containing list field is invalid', () => {
+        const config = {
+            filename: '${bananaBrands}.md',
+            fields: [
+                {name: 'version', type: 'semver'},
+                {name: 'bananaBrands', type: 'string', list: true},
+            ],
+            standard: {
+                query: '',
+                sort: {
+                    field: 'version',
+                    order: 'desc',
+                },
+            },
+        };
+        expect(parseConfig(config)).toMatchObject(
+            left({msg: 'Error in filename - Fields used in the file name must not be optional or lists: bananaBrands'})
```

Comment:
Error Nachricht ist einwenig gedoppelt. Wie wäre es mit `error in fieldname: Referenced field 'bananaBrands' does not exist` oder  `error in fieldname: Referenced field 'bananaBrands' is optional. Only required fields may be used.` oder `error in fieldname: Referenced field 'bananaBrands' is type list. Only non list types may be used.`

Reply:
filename statt fieldname, aber ansonsten find ich das ganz gut =)